dashboard/views.py

```diff
@@ -213,7 +213,7 @@ def grade_distribution(request, course_id=0):
     grade_score_sql = "select current_grade,(select current_grade from user where sis_name=" \
                       "%(current_user)s and course_id=%(course_id)s) as current_user_grade from user where course_id=%(course_id)s;"
     df = pd.read_sql(grade_score_sql, conn, params={"current_user": current_user,'course_id': course_id})
-    if df.empty:
+    if df.empty or df['current_grade'].isnull().all():
```

Comment:
Nice catch! all 3 courses have grades come in in my local DB, so I couldn't catch this bug

Reply:
Yeah thanks I noticed a stack trace locally when there were no grades for this user at all. This caught that case.

Reply:
I feel the stack trace would have occurred when there are no grades at all for the course vs no grade for user. The code is filtering out the null's and if user grade is null then it is shown as message of "no grades for you"

Reply:
Yeah, that might have been what happened too. Basically if the nulls are all filtered out and df was empty after that process was done. This just catches it a little bit earlier.